apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryHandler.cs

```diff
@@ -69,14 +74,11 @@ public async Task<bool> HandleResponseAsync(HandleUnsuccessfulResponseArgs args)
             {
                 return false;
             }
-            // The first failure will have args.CurrentFailedTry set to 1,
-            // whereas we want the first delay to be 1 second. We use Math.Min on the power
-            // rather than on the result to obtain a max retry of 32 seconds without risking
-            // calling Math.Pow with a huge number.
-            int power = Math.Min(args.CurrentFailedTry - 1, 5);
-            double seconds = Math.Pow(2.0, power);
-            var delay = TimeSpan.FromSeconds(seconds);
-            await Task.Delay(delay, args.CancellationToken).ConfigureAwait(false);
+
+            _backoffMultiplier = (_backoffMultiplier == 0) ? Math.Pow(2.0, args.CurrentFailedTry - 1) : _backoffMultiplier;
+            TimeSpan delay = _initialBackoff + TimeSpan.FromSeconds((args.CurrentFailedTry - 1) * _backoffMultiplier);
```

Comment:
We have initial backoff, we have max backoff, we have backoff multiplier. We just need to determine the next delay based on `CurrentFailedTry` . Now can we make it modular and clean?